src/pybind/mgr/dashboard/module.py

```diff
@@ -248,15 +247,51 @@ class Module(MgrModule, CherryPyConfig):
     PLUGIN_MANAGER.hook.register_commands()
 
     MODULE_OPTIONS = [
-        {'name': 'server_addr'},
-        {'name': 'server_port'},
-        {'name': 'jwt_token_ttl'},
-        {'name': 'password'},
-        {'name': 'url_prefix'},
-        {'name': 'username'},
-        {'name': 'key_file'},
-        {'name': 'crt_file'},
-        {'name': 'ssl'}
+        {
+            'name': 'server_addr',
+            'type': 'str',
+            'default': '::'
+        },
+        {
+            'name': 'server_port',
+            'type': 'int',
+            'default': 8443  # SSL is enabled by default
+        },
+        {
```

Comment:
If we add 2 separate options for regular and HTTPS ports, we may leave CephMgr take care of the defaults. Most servers I worked with used this approach: either they used a fixed port for both or separate variables (`port` and `ssl_port`).

```suggestion
        {
            'name': 'server_ssl_port',
            'type': 'int',
            'default': 8443,
        },
        {        
```

Reply:
This seems to be a good idea, but needs more work because the QA test framework needs to be adapted in this case. See https://github.com/ceph/ceph/blob/master/qa/tasks/mgr/dashboard/helper.py#L113.
Because of that this should be done in a separate PR. I've opened a new [issue](https://tracker.ceph.com/issues/38331) for that.